pkg/scheduler/queue/tenant_queues.go

```diff
@@ -155,10 +162,24 @@ func (qb *queueBroker) enqueueRequestFront(request *tenantRequest, tenantMaxQuer
 		return err
 	}
 
-	queuePath := QueuePath{string(request.tenantID)}
+	queuePath, err := qb.makeQueuePath(request)
+	if err != nil {
+		return err
+	}
 	return qb.tenantQueuesTree.EnqueueFrontByPath(queuePath, request)
 }
 
+func (qb *queueBroker) makeQueuePath(request *tenantRequest) (QueuePath, error) {
+	if qb.additionalQueueDimensionsEnabled {
+		if schedulerRequest, ok := request.req.(*SchedulerRequest); ok {
+			return append(QueuePath{string(request.tenantID)}, schedulerRequest.AdditionalQueueDimensions...), nil
+		}
+	}
```

Comment:
context from slack, just decided to leave it as is for now:

> 
> We don’t technically need it, but I had the separate flag because:
> a)  it seemed odd to just have the flag in the frontend when all the important/complicated stuff is happening in the scheduler and
> b) It just seemed more proper/less coupled? If something needs troubleshooting, we could use on flag to stop sending extra dimensions from the frontend or another flag for the scheduler to just ignore the extra dimensions.
> 
> Don’t feel strongly though. I scanned the config docs and there are other multi-component features that need a flag on both components to work.